models/setup.py

```diff
@@ -0,0 +1,40 @@
+import pip
+import os
+import sys
+import platform
+
+# pip 10.0 introduced a breaking change that moves the location of main
+try:
+    from pip import main
+except ImportError:
+    from pip._internal import main
+
+
+def install_dependencies(packages):
+    pip_arg = ['install'] + packages + ['--no-cache-dir']
+    if hasattr(pip, 'main'):
+        pip.main(pip_arg)
+    else:
+        pip._internal.main(pip_arg)
+
+
+if __name__ == '__main__':
+    install_dependencies(['sklearn', 'pandas', 'numpy',
```

Comment:
So I'd generally advise against having multiple setup files like this within your application. It is entirely possible at certain points in times that dependencies here may conflict with the overall application dependencies, so doing it in this fashion could upgrade / downgrade packages in a way that's not intuitive to end users.

You might alternately want to consider using a requirements file for conda which has optional dependencies and make sure that users install optional ones if they want these models. It would be a rather large scope and probably worth a separate PR from this but I think nonetheless would make things easier

Reply:
Yeah you're completely right and it's definitely a known risk. However, since these pre-deployed models are an optional feature and the packages aren't installed by default we chose to implement them this way for the time being.